File: ios/MyWatch Watch App/Todo.swift
//
//  Todo.swift
//  Runner
//
//  Created by Koichi Kishimoto on 2024/10/23.
//

struct Todo: Identifiable {
    var id: String
    var title: String
    var description: String
    var isDone: Bool
    
    init?(dictionary: [String: Any]) {
        guard let id = dictionary["id"] as? String,
              let title = dictionary["title"] as? String,
              let description = dictionary["description"] as? String,
              let isDone = dictionary["isDone"] as? Bool else {
            return nil
        }
        self.id = id
        self.title = title
        self.description = description
        self.isDone = isDone
    }
}

File: lib/pigeons/todo.dart
import 'package:pigeon/pigeon.dart';

@ConfigurePigeon(PigeonOptions(
  dartOut: 'lib/todo/todo.g.dart',
  dartOptions: DartOptions(),
  kotlinOut:
      'android/app/src/main/kotlin/dev/flutter/pigeon_example_app/Todo/Todo.g.kt',
  kotlinOptions: KotlinOptions(),
  swiftOut: 'ios/Runner/Todo/Todo.g.swift',
  swiftOptions: SwiftOptions(),
))

class Todo {
  Todo(this.id, this.title, this.description, this.isDone);

  String id;
  String title;
  String description;
  bool isDone;
}

@FlutterApi()
abstract class TodoFlutterApi {
  void addTodo(Todo todo);
  void deleteTodo(Todo todo);
  @async
  List<Todo?> fetchTodos();
}
